### PR TITLE
Feat like history

### PR DESCRIPTION
**Pull request by Kothalkarpriya**
I have created a Like /Dislike Feature and History Feature.
When we like a video that gets added to the Like page and similarly when we visit the specific video page it will get added to history. Their are improvements to be made

### CODE REVIEW

**Review by neetukumari4858 — APPROVED** [review]
code is readable 
Naming convention is also good
code Structure looks  good

File: src/backend/utils/historyReducer.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      return {
+        ...state,
+        historyList: [
+          ...state.historyList.filter((item) => item.id !== action.payload.id),
+        ],
+      };
+    case "CLEAR_HISTORY":
+      return {
+        ...state,
+        historyList: [],
```
Comment (tusharkandpal) [inline src/backend/utils/historyReducer.jsx:20]: you can provide the initial state here.

File: src/components/AddToWatchBtns.jsx [hunk 2]
```diff
@@ -1,24 +1,59 @@
...
+          >
+            <AiFillLike className="btn-icon" /> Like
+          </button>
+        ) : (
+          <button
+            className="btn"
+            onClick={() =>
+              dispatchLike({
+                type: "LIKE_DEL",
+                payload: defaultLike,
```
Comment (tusharkandpal) [inline src/components/AddToWatchBtns.jsx:49]: would recommend you to pass `payload` as object

**Review by JantuDeb — APPROVED** [review]
Overall this is looking good

**Review by tusharkandpal — APPROVED** [review]
Looks good to merge .! suggested minor changes

File: src/components/AddToWatchBtns.jsx (hunk 2 above)
Comment (Kothalkarpriya) [reply src/components/AddToWatchBtns.jsx:49]: it is already in a object

File: src/backend/utils/historyReducer.jsx (hunk 1 above)
Comment (Kothalkarpriya) [reply src/backend/utils/historyReducer.jsx:20]: ok

**Pull request merged**